### PR TITLE
local-dev: pull images on every run + add support for tty resizing

### PR DESCRIPTION
**Pull request by ayushkamat**

### CODE REVIEW

File: latch_cli/services/local_dev.py [hunk 1]
```diff
@@ -233,22 +266,32 @@ def file_filter(filename: str) -> bool:
...
-                                return
-
-                            if cmd.startswith("run"):
-                                await aioconsole.aprint(
-                                    "Syncing your local changes... "
-                                )
+                                raise EOFError
+
+                            if not COMMAND_EXPRESSION.match(cmd):
+                                await aioconsole.aprint("Invalid command")
```
Comment (kennyworkman) [inline latch_cli/services/local_dev.py:272]: parsing out useful errors from a failed match would be instructive here (eg. run-scripts vs run-script often trips me up)

File: latch_cli/services/local_dev.py [hunk 2]
```diff
@@ -233,22 +266,32 @@ def file_filter(filename: str) -> bool:
...
+                                continue
+
+                            if (
+                                cmd.startswith("run")
```
Comment (kennyworkman) [inline latch_cli/services/local_dev.py:276]: you should actually be able to include arbitrary whitespace characters before your first command and still get desired behavior (try with bash)

File: latch_cli/services/local_dev.py [hunk 3]
```diff
@@ -33,10 +38,41 @@
 EXEC_COMMANDS = ["shell", "run", "run-script"]
 LIST_COMMANDS = ["ls", "list-tasks"]
 
+COMMAND_EXPRESSION = re.compile("^(run\s.+|run-script\s.+|shell|list-tasks|ls.*)$")
```
Comment (kennyworkman) [inline latch_cli/services/local_dev.py:41]: you should actually be able to include arbitrary whitespace characters before your first command and still get desired behavior (try with bash)

File: latch_cli/services/local_dev.py [hunk 4]
```diff
@@ -301,32 +346,49 @@ async def shell_session(
         loop,
     )
 
+    resize_event_queue = asyncio.Queue()
+
+    def new_sigwinch_handler(signum, frame):
+        if isinstance(old_sigwinch_handler, Callable):
```
Comment (kennyworkman) [inline latch_cli/services/local_dev.py:352]: would handle the cases when [you dont get a callable](https://docs.python.org/3/library/signal.html#signal.getsignal)

Comment (kennyworkman) [reply latch_cli/services/local_dev.py:352]: i also remember there is a builtin to do this - https://docs.python.org/3/library/functions.html#callable, but i actually can't think of why this is better

i'm pretty sure `isinstance(obj, Callable) iff obj has __call__` which is exactly what builtin does, literally cant find anything else about it

**Review by kennyworkman — CHANGES_REQUESTED** [review]

File: latch_cli/services/local_dev.py [hunk 5]
```diff
@@ -33,10 +38,41 @@
...
 COMMANDS = QUIT_COMMANDS + EXEC_COMMANDS + LIST_COMMANDS
 
 
-async def copy_files(ssh_conn: asyncssh.SSHClientConnection, pkg_root: Path):
+def get_latest_image(pkg_root: Path) -> str:
+    import_flyte_objects(paths=[pkg_root])
+
+    wf_name = ""
+    for entity in FlyteEntities.entities:
+        if isinstance(entity, PythonFunctionWorkflow):
```
[the remark this answers is not in the document]
Comment (ayushkamat) [reply latch_cli/services/local_dev.py:51]: they would all have the same version, so not an issue

File: latch_cli/services/local_dev.py [hunk 6]
```diff
@@ -33,10 +38,41 @@
...
+        json={
+            "wf_name": wf_name,
+            "ws_account_id": current_workspace(),
+        },
+    )
+
+    resp.raise_for_status()
+    latest_version = resp.json()["version"]
+
+    return f"{config.dkr_repo}/{current_workspace()}_{pkg_root.name}:{latest_version}"
```
[the remark this answers is not in the document]
Comment (ayushkamat) [reply latch_cli/services/local_dev.py:69]: we can add it to the backlog lol

File: latch_cli/services/local_dev.py [hunk 7]
```diff
@@ -78,16 +117,36 @@ async def get_messages(ws: websockets.WebSocketClientProtocol, show_output: bool
 async def send_message(
     ws: websockets.WebSocketClientProtocol,
     message: Union[str, bytes],
+    typ: Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (ayushkamat) [reply latch_cli/services/local_dev.py:120]: good call

File: latch_cli/services/local_dev.py (hunk 1 above)
Comment (ayushkamat) [reply latch_cli/services/local_dev.py:272]: also a good call

File: latch_cli/services/local_dev.py (hunk 2 above)
Comment (ayushkamat) [reply latch_cli/services/local_dev.py:276]: fixed

File: latch_cli/services/local_dev.py (hunk 4 above)
Comment (ayushkamat) [reply latch_cli/services/local_dev.py:352]: the cases where i dont get a callable arent things i dont need to handle - SIG_IGN is literally ignore the signal in this process, SIG_DFL for the signal im handling here (SIGWINCH) means the current handler does the default action for SIGWINCH, which is also to ignore it (https://www.gnu.org/software/libc/manual/html_node/Miscellaneous-Signals.html), and None ofc means that there is no handler for it

**Pull request merged**